dialoptions.go

```diff
@@ -318,6 +321,8 @@ func WithBlock() DialOption {
 //
 // Notice: This API is EXPERIMENTAL and may be changed or removed in a
 // later release.
```

Comment:
Could you please remove this experimental warning and we'll just call it supported now ("throughout 1.x")?

Reply:
PTAL https://github.com/grpc/grpc-go/pull/7097/commits/cce30008634fa201fb6d8500b9c12f6b3c806d04